src/legacy/core_plugins/data/public/search/aggs/test_helpers/mock_data_services.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { coreMock } from '../../../../../../../../src/core/public/mocks';
+import { dataPluginMock } from '../../../../../../../plugins/data/public/mocks';
+import { searchStartMock } from '../../mocks';
+import { setSearchServiceShim } from '../../../services';
+import {
+  setUiSettings,
+  setQueryService,
+  setIndexPatterns,
+  setFieldFormats,
+  setSearchService,
+  setOverlays,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../../../../../plugins/data/public/services';
+
+/**
+ * Testing helper which calls all of the service setters used in the
+ * data plugin. Services are added using their provided mocks.
+ *
+ * @internal
+ */
+export function mockDataServices() {
```

Comment:
I don't love that we need to do something like this in order to test services that use getters/setters. Open to suggestions -- perhaps we do this in the main data mocks, so when someone calls `dataPluginMock.createStartServices()`, this already happens for them in the background?

Would prefer a different long-term solution, but okay with this in the interim if it keeps us moving forward.